lib/anubis/server/session/supervisor.ex

```diff
@@ -91,7 +100,64 @@ defmodule Anubis.Server.Session.Supervisor do
   end
 
   @impl DynamicSupervisor
-  def init(_init_arg) do
+  def init({_server, _registry}) do
     DynamicSupervisor.init(strategy: :one_for_one)
   end
+
+  # Private functions
+
+  defp restore_sessions(server, registry) do
+    case get_store() do
+      nil ->
+        Anubis.Logging.log(:debug, "No session store configured, skipping session restoration", [])
+
+        :ok
+
+      store ->
+        Anubis.Logging.log(:debug, "Checking for sessions to restore from store", server: server)
+
+        case store.list_active(server: server) do
+          {:ok, session_ids} ->
+            if length(session_ids) > 0 do
+              Anubis.Logging.log(:info, "Restoring sessions", count: length(session_ids), server: server)
+
+              Enum.each(session_ids, fn session_id ->
+                Anubis.Logging.log(:debug, "Creating session process for restored session", session_id: session_id)
+
+                create_session(registry, server, session_id)
+              end)
+            else
+              Anubis.Logging.log(:debug, "No sessions found to restore for server", server: server)
+            end
+
+            :ok
+
+          {:error, reason} ->
+            Anubis.Logging.log(:warning, "Failed to list active sessions from store", server: server, reason: reason)
+
+            :ok
+        end
+    end
+  end
+
+  defp get_store do
+    case Application.get_env(:anubis_mcp, :session_store) do
```

Comment:
i think this application config fetching should be either on `Anubis` module or a new `Anubis.Config` one to abstract this config fetching and logic

also, this got very complex with multiple nesting levels, we can simplify it with something like

```elixir
config = Application.get_env(:anubis_mcp, :session_store)
enabled? = config[:enabled] || false
adapter = config[:adapter]

if enabled? && Code.ensure_loaded?(adapter), do: adapter
```

`Code.ensure_loaded?/1` when given `nil` always returns `false`